### PR TITLE
Document rpc-config until skip-service-descriptor

### PR DESCRIPTION
**Pull request by ben-grande**
Fixes: https://github.com/QubesOS/qubes-issues/issues/9036

See comment https://github.com/QubesOS/qubes-issues/issues/9036#issuecomment-2083760576

### CODE REVIEW

File: qubes-rpc-config/README [hunk 1]
```diff
@@ -1,13 +1,36 @@
...
+* force-user: Enforce that service should be run by the specified username.
+  Useful to set the user at the target qube where it can be verified that it
+  exists instead of having to modify the policy in dom0.
+  Accepted values: string
+
+* wait-for-session: Wait for full GUI session initialization before starting
+  the service. Controlled by the RPC service qubes.WaitForSession. There is
+  no timeout, if the session is never initialized (for example because there
+  is no GUI running at all), service will never be started.
+  Accepted values: true (default), false, 0 and 1 accepted for backwards
```
Comment (marmarek) [inline qubes-rpc-config/README:29]: The default is "false".
(I guess got this from `qvm-run`, which by default uses a service that has `wait-for-session=true` set)

File: qubes-rpc-config/README [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
-  (for example because there is no GUI running at all), service will never be
-  started.
-  Allowed values are 0 or 1.
+* force-user: Enforce that service should be run by the specified username.
+  Useful to set the user at the target qube where it can be verified that it
+  exists instead of having to modify the policy in dom0.
+  Accepted values: string
+
+* wait-for-session: Wait for full GUI session initialization before starting
+  the service. Controlled by the RPC service qubes.WaitForSession. There is
```
Comment (marmarek) [inline qubes-rpc-config/README:26]: "Controlled" is a weird word for this, maybe "Implemented" or "Performed"?

File: qubes-rpc-config/README [hunk 3]
```diff
@@ -1,13 +1,36 @@
...
+  no timeout, if the session is never initialized (for example because there
+  is no GUI running at all), service will never be started.
+  Accepted values: true (default), false, 0 and 1 accepted for backwards
+  compatibility.
 
+* skip-service-descriptor: Skip sending service descriptor and go for the
```
Comment (marmarek) [inline qubes-rpc-config/README:32]: Maybe worth adding a link to https://www.qubes-os.org/doc/qrexec-socket-services/ for explanation what the service descriptor is?

File: qubes-rpc-config/README [hunk 4]
```diff
@@ -1,13 +1,36 @@
...
+  actual data directly. Useful to skip sending metadata to socket-based
+  services, thus the service must be a socket, not an executable.
```
Comment (marmarek) [inline qubes-rpc-config/README:34]: I'd say "thus the option is valid only for a socket services, not an executable"

**Review by marmarek — CHANGES_REQUESTED** [review]

**Comment by ben-grande** [comment]
I understand it is not markdown, but it is easier to read if there are bulleted lists, of which, I followed [Google's markdown guide for nested-list-spacing](https://google.github.io/styleguide/docguide/style.html#nested-list-spacing).

Changes requested above were resolved.

File: qubes-rpc-config/README [hunk 5]
```diff
@@ -1,13 +1,53 @@
...
+    sign (=) between the key and the value, while any number of spaces or tabs
+    from the key to the equal sign or from the equal sign to the value are
+    also supported, e.g. "key=value", "key   =  value", "key	=
+    value".
+
+*   Comments: Lines starting with hashtag/octothorpe/pound sign "#" are
+    ignored.
+
+*   Boolean values: true, false, 0, 1. For backwards compatibility, only the
+    options "wait-for-session" and "skip-service-descriptor" accepts the
```
Comment (DemiMarie) [inline qubes-rpc-config/README:17]: Only `wait-for-session` accepts it.  `skips-service-descriptor` does not.

File: qubes-rpc-config/README [hunk 6]
```diff
@@ -1,13 +1,53 @@
...
+    *   Default value: false
+    *   Example: wait-for-session=true
 
+*   skip-service-descriptor:
+    *   Description: Skip sending service descriptor and go for the actual
+        data directly. Useful to skip sending metadata to socket-based
+        services, thus the option is valid only for socket services, not
+        executables. See https://www.qubes-os.org/doc/qrexec-socket-services/
+    *   Value type: boolean
+    *   Accepted values: true, false, 0, 1.
```
Comment (DemiMarie) [inline qubes-rpc-config/README:51]:
```suggestion
    *   Accepted values: true, false.
```

File: qubes-rpc-config/README [hunk 7]
```diff
@@ -1,13 +1,53 @@
...
+    *   Value type: string
+    *   Default value: same user as in the policy, else it is 'user'.
+    *   Example: 'user'
+
+*   wait-for-session:
+    *   Description: Wait for full GUI session initialization before starting
+        the service. Implemented by the RPC service qubes.WaitForSession.
+        There is no timeout, if the session is never initialized (for example
+        because there is no GUI running at all), service will never be
+        started.
```
Comment (DemiMarie) [inline qubes-rpc-config/README:39]:
```suggestion
        started. It is possible for the service to be started even
        if qubes.WaitForSession fails, so the service must not depend on
        wait-for-session=true for security.
```

**Review by DemiMarie — CHANGES_REQUESTED** [review]

File: qubes-rpc-config/README [hunk 8]
```diff
@@ -1,13 +1,36 @@
...
+* Configuration format: key value pair that must be separated by an equal sign
+  '=' between the key and the value, while any number of spaces or tabs from
+  the key to the equal sign or from the equal sign to the value are also
+  supported, e.g. "key=value", "key   =  value", "key	=	value".
+
+* Comments: Lines starting with hashtag/octothorpe/pound sign "#" are ignored.
+
+* Boolean values: "true" and "false".
+
+* String values: must be enclosed by single quotes "'", e.g. 'str'.
```
[the remark this answers is not in the document]
Comment (ben-grande) [reply qubes-rpc-config/README]: Can we instead document which characters are supported? Or is it easier to say what is not supported? Pushing a commit with your suggestion for now.

**Review by marmarek — APPROVED** [review]

**Pull request merged**